plugins/destination/mongodb/client/migrate.go

```diff
@@ -61,14 +60,14 @@ func (c *Client) migrateTableOnConflict(ctx context.Context, table *arrow.Schema
 	return nil
 }
 
-func (c *Client) getIndexTemplates(table *arrow.Schema) []mongo.IndexModel {
+func (c *Client) getIndexTemplates(table *schema.Table) []mongo.IndexModel {
 	var indexes []mongo.IndexModel
 
-	pks := schema.PrimaryKeyIndices(table)
+	pks := table.PrimaryKeysIndexes()
```

Comment:
```suggestion
	pks := table.PrimaryKeys()
```
As this will actually give you the column names already